lib/assert.js

```diff
@@ -931,6 +933,75 @@ assert.doesNotMatch = function doesNotMatch(string, regexp, message) {
   internalMatch(string, regexp, message, doesNotMatch);
 };
 
+const noop = () => {};
+
+const mustCallChecks = [];
+
+function runCallChecks(exitCode) {
+  if (exitCode !== 0) return;
```

Comment:
Maybe we should only do this after printing the messages? This might be a bit surprising otherwise, not seeing the messages printed when the function’s weren’t called the right number of times?

Reply:
I'm not sure if I understand what you're referencing, is it the line `if (exitCode !== 0) return;` that should be done after the messages are printed